src/h5web/explorer/Explorer.tsx

```diff
@@ -18,25 +18,53 @@ function Explorer(props: Props): JSX.Element {
 
   const [selectedNode, setSelectedNode] = useState<TreeNode<HDF5Link>>();
 
+  useEffect(() => {
+    if (tree) {
+      // Select root node when tree is ready
+      setSelectedNode(tree);
+    }
+  }, [tree]);
+
   useEffect(() => {
     if (selectedNode) {
+      // Propagate selected link to parent component
       onSelect(selectedNode.data);
     }
   }, [selectedNode, onSelect]);
 
+  if (!tree) {
```

Comment:
While the tree is loading (i.e. while the metadata is being fetched and the tree is being built), I render the domain and a loading text right away.